engine/src/player-platform.cpp

```diff
@@ -1308,6 +1308,9 @@ MCPlayerDuration MCPlayer::getmoviecurtime()
 
 void MCPlayer::setcurtime(MCPlayerDuration newtime, bool notify)
 {
+    MCPlatformPlayerDuration t_duration = getduration();
+    if (newtime > t_duration)
+        newtime = t_duration;
```

Comment:
Using MCMax would be more succinct here:
`newtime = MCMax(newtime, getduration());`